api/src/main/java/io/strimzi/api/kafka/model/KafkaListenerAuthenticationScramSha512.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+
+/**
+ * Configures a listener to use SASL SCRAM-SHA-512 for authentication.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = true,
+        builderPackage = "io.strimzi.api.kafka.model"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class KafkaListenerAuthenticationScramSha512 extends KafkaListenerAuthentication {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String SCRAM_SHA_512 = "scram-sha-512";
```

Comment:
Hmm, I'm not sure I like the `scram-sha-512` as the type. But it is probably the most future proof version I guess.

Reply:
We can either put the hash algo and length in the `type` name (as I've done here) or have an  object with `type: scram` which we then have to configure with hash algo and length. Since we currently only plan to support SHA-512 there doesn't seem to be much point in having an extra level, since this thing is already pretty deeply nested.

Reply:
I think parametrizing this is unnecessary. Its not like this is freely configurable anyway. You can just choose 256 or 512. I was more thinking whether having `type: scram` wouldn't be enough. But we never know when someone comes and says he want to use 256 for some reason I guess.